### PR TITLE
add canDelete property to the token JSON object fix for #27311

### PR DESCRIPTION
**Pull request by individual-it**
## Description

adds canDelete property to the token JSON object

## Related Issue

#27311

## How Has This Been Tested?

selenium tests written
1 line fix >200 lines of test :-)

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: lib/private/Authentication/Token/DefaultToken.php [hunk 1]
```diff
@@ -111,6 +111,7 @@ public function jsonSerialize() {
 			'name' => $this->name,
 			'lastActivity' => $this->lastActivity,
 			'type' => $this->type,
+			'canDelete' => true,
```
Comment (PVince81) [inline lib/private/Authentication/Token/DefaultToken.php]: Can you explain why this change is needed in the real code for tests to run ?
I don't see this used in the PR, so not sure.

Comment (individual-it) [inline lib/private/Authentication/Token/DefaultToken.php]: that is the fix for #27311 itself.
The delete button is not shown directly after adding a password because the JS object does not have the "canDelete" property set: https://github.com/owncloud/core/blob/master/settings/js/panels/authtoken_view.js#L32
it is not set because the Ajax in https://github.com/owncloud/core/blob/master/settings/js/panels/authtoken_view.js#L182 does not return any value for that property
So this line does add the "canDelete" property to a newly created password

If you reload the page the "canDelete" is set by this line https://github.com/owncloud/core/blob/master/settings/Controller/AuthSettingsController.php#L102

**Comment by PVince81** [comment]
Ok, got it 👍
